dhall/src/Dhall/Import.hs

```diff
@@ -554,7 +555,7 @@ writeToSemanticCache :: Dhall.Crypto.SHA256Digest -> Data.ByteString.ByteString
 writeToSemanticCache hash bytes = do
     _ <- Maybe.runMaybeT $ do
         cacheFile <- getCacheFile "dhall" hash
-        liftIO (Data.ByteString.writeFile cacheFile bytes)
+        liftIO (System.AtomicWrite.Writer.ByteString.atomicWriteFile cacheFile bytes)
```

Comment:
I'm not yet sure whether we ought to use

http://hackage.haskell.org/package/atomic-write-0.2.0.6/docs/System-AtomicWrite-Writer-ByteString.html#v:atomicWriteFileWithMode

with some "binary" `FileMode` instead. I can't actually find any documentation on constructing these `FileMode`s.

Reply:
Ah no, `FileMode` seems to be about file permissions instead of text mode vs binary mode.

Reply:
I've asked for clarification regarding newline adjustments in https://github.com/stackbuilders/atomic-write/issues/16.

Reply:
@sjakobi: Yeah, I don't believe binary mode is possible using the current `atomic-write` API

Reply:
I'll wait a bit for feedback from the `atomic-write` maintainers. Not sure whether I should then make a patch to add binary mode writes to `atomic-write`, or just switch to `unliftio` (which however doesn't seem to implement atomicity on Windows).

It would be nice if we could use the same package to fix https://github.com/dhall-lang/dhall-haskell/issues/498 too…

Reply:
I have opened https://github.com/fpco/unliftio/issues/50 regarding Windows support and https://github.com/fpco/unliftio/issues/51 regarding support for text mode in `unliftio`.

Reply:
Feel free to switch to `unliftio` if that ends up being easier

Reply:
`atomic-write` now offers binary mode writes too, which I'm using now.